### DISCUSSION TITLE
Linking filter graph

### DISCUSSION THREAD

**Discussion by anvlkv** [zmwangx/rust-ffmpeg#145] [Q&A]
Hi, I’m trying to create a concat filter. But seems I’m doing something wrong. Or maybe its the avfilter_graph_parse_ptr?

create a concat filter which should accept 3x2 inputs

```
filter.add(&filter::find("buffersink").unwrap(), "out_v", "").unwrap();
filter.add(&filter::find("abuffersink").unwrap(), "out_a", "").unwrap();
filter.add(&filter::find("buffer").unwrap(), "in_v_0", &args_v0).unwrap();
filter.add(&filter::find("buffer").unwrap(), "in_v_1", &args_v1).unwrap();
filter.add(&filter::find("buffer").unwrap(), "in_v_2", &args_v2).unwrap();
filter.add(&filter::find("abuffer").unwrap(), "in_a_0", &args_a0).unwrap();
filter.add(&filter::find("abuffer").unwrap(), "in_a_1", &args_a1).unwrap();
filter.add(&filter::find("abuffer").unwrap(), "in_a_2", &args_a2).unwrap();
filter.input("out_v", 0).unwrap()
   .input("out_a", 0).unwrap()
   .output("in_v_0", 0).unwrap()
   .output("in_v_1", 0).unwrap()
   .output("in_v_2", 0).unwrap()
   .output("in_a_0", 0).unwrap()
   .output("in_a_1", 0).unwrap()
   .output("in_a_2", 0).unwrap()
   .parse("[in_v_0][in_a_0][in_v_1][in_a_1][in_v_2][in_a_2]concat=n=3:v=1:a=1[out_v][out_a]")
   .unwrap();
filter.validate().expect("invalid filter");
```

My validate call fails with: Output pad "default" with type audio of the filter instance "in_a_0" of abuffer not connected to any destination

I've tried using avfilter_graph_parse2 and adding all my sinks and buffers to the spec:

buffer=width=1280:height=720:pix_fmt=yuv420p:time_base=1/15360:sar=1[in_v_0];
abuffer=time_base=1/48000:sample_rate=48000:sample_fmt=fltp:channel_layout=0x3[in_a_0];
buffer=width=1280:height=720:pix_fmt=yuv420p:time_base=1/15360:sar=1[in_v_1];
abuffer=time_base=1/48000:sample_rate=48000:sample_fmt=fltp:channel_layout=0x3[in_a_1];
buffer=width=1280:height=720:pix_fmt=yuv420p:time_base=1/15360:sar=1[in_v_2];
abuffer=time_base=1/48000:sample_rate=48000:sample_fmt=fltp:channel_layout=0x3[in_a_2];
[in_v_0][in_a_0][in_v_1][in_a_1][in_v_2][in_a_2]concat=n=3:v=1:a=1[out_v][out_a];[out_v]buffersink;
[out_a]abuffersink
Which then fails with Input pad "default" with type video of the filter instance "out_v" of buffersink not connected to any source

I hope someone can point out what I’m doing so wrong.

**Comment by joprice** [comment]
I have the same issue. The transcoder example works for me, but an example of building a more complex graph would be great.

**Comment by joprice** [comment]
@anvlkv I put up a fix for this https://github.com/zmwangx/rust-ffmpeg/pull/156 that fixed the issue for me.
